nestkernel/node_manager.cpp

```diff
@@ -308,6 +311,28 @@ NodeManager::add_music_nodes_( Model& model, index min_node_id, index max_node_i
   } // omp parallel
 }
 
+NodeCollectionPTR
+NodeManager::node_id_to_nodeCollection( const index node_id ) const
+{
+  auto it = std::lower_bound( nodeCollection_last_.begin(), nodeCollection_last_.end(), node_id );
```

Comment:
Please add a comment explaining the logic here. It took me some minutes to get it ...